---
title: "R Notebook"
output:
  html_document:
    df_print: paged
    toc: true
    toc_float: 
      smooth_scroll: true
---

```{r}
library(Seurat)

seurobj <- readRDS('/projects/pytrik/sc_adipose/analyze_10x_fluidigm/data/10x-180831')
aligned <- readRDS('/projects/pytrik/sc_adipose/analyze_10x_fluidigm/data/10x-180831-aligned')
aligned_T1_old <- readRDS('/projects/pytrik/sc_adipose/analyze_10x_fluidigm/data/10x-180504-T1-aligned')
```

#Quality control plots

For now the data was only filtered on 0.1 percent.mito. The subcluster from T1 with low UMI and gene counts was removed.

```{r fig1, fig.height=15, fig.width=10, fig.align="center"}
VlnPlot(seurobj, c("nGene", "percent.mito", "nUMI"), group.by='timepoint', nCol = 1, point.size.use=-1, size.x.use = 10)
```

```{r}
GenePlot(seurobj, 'nUMI', 'nGene', cex.use = 0.5)
```


```{r}
PCElbowPlot(seurobj, num.pc=50) #The elbow is at 6 pcs. Run tSNE on both 6, 18 and 30 PC's. 
```

#Data visualization

Interesting to see: T4 and T5 contain a lot more variation than T1, T2 and T3, and PC2 seems to split T4 and T5 (most cells getting a negative PC2 score, but some cells getting a positive PC2 score). Could the split in PC2 describe the cells developing into white or brown? 

```{r}
PCAPlot(seurobj, group.by='timepoint', pt.size=0.1)
```

If EBF2 marks brown then not. 

```{r}
FeaturePlot(seurobj, reduction.use='pca', features.plot='EBF2', pt.size=1, cols.use=c('gray', 'blue'), no.legend=F)
```


TSNE on 6 PC's. 

```{r}
DimPlot(seurobj, reduction.use='tsne6', group.by='timepoint', pt.size=0.1)
```

The first 6 PCs explain most of the variation (as we could see in the elbow plot above), but they don't capture the more subtle differences. For example, when only using 6 PC's the mixture cluster cannot be found:

```{r}
DimPlot(seurobj, reduction.use='tsne6', pt.size=0.1, cells.highlight=seurobj@cell.names[seurobj@meta.data$res.0.5 == 11], cols.use='grey', cols.highlight='blue')
```

TSNE on 18 PC's. 

```{r}
DimPlot(seurobj, reduction.use='tsne18', group.by='timepoint', pt.size=0.1)
```

```{r}
DimPlot(seurobj, reduction.use='tsne18', group.by='Phase', pt.size=0.1)
```

Cluster 11 = mixture cluster.

```{r}
DimPlot(seurobj, reduction.use='tsne18', group.by='res.0.5', pt.size=0.1)
```

#Metadata feature plots

```{r}
FeaturePlot(seurobj, reduction.use='tsne18', features.plot = 'nUMI', cols.use=c('grey', 'blue'), no.legend=F)
```

```{r}
FeaturePlot(seurobj, reduction.use='tsne18', features.plot = 'percent.mito', cols.use=c('grey', 'blue'), no.legend = F)
```

```{r}
FeaturePlot(seurobj, reduction.use='tsne18', features.plot = 'nGene', cols.use=c('grey', 'blue'), no.legend = F)
```

#Alignment

Shared correlation per CC.

```{r}
load('/projects/pytrik/sc_adipose/analyze_10x_fluidigm/data/MetageneBicorPlots')
ccplot_aligned
```

12 CC's were used for tSNE.

```{r}
TSNEPlot(aligned, group.by='timepoint', pt.size=0.1)
```

```{r}
TSNEPlot(aligned, group.by='Phase', pt.size=0.1)
```

#Aligning T1 with old data

Shared correlation per CC. 

```{r}
#ccplot_aligned_T1_old  make new plot
```

15 CC's were used for tSNE. 

```{r}
#replacing res.0.5 column of T1 with 'T1'
aligned_T1_old@meta.data[aligned_T1_old@meta.data$sample_name == 'T1', 'res.0.5'] = 'T1'

TSNEPlot(aligned_T1_old, group.by='sample_name', pt.size=0.1)
```

```{r}
TSNEPlot(aligned_T1_old, group.by='res.0.5', pt.size=0.1)
```

Highligted mixture cluster

```{r}
DimPlot(aligned_T1_old, reduction.use='tsne', cells.highlight=aligned_T1_old@cell.names[which(aligned_T1_old@meta.data$res.0.5 %in% 12)], pt.size.use=0.1, cols.highlight = 'blue', cols.use='grey')
```

Highlighted T1 cells. It does not look like T1 cells are in the mixture cluster on the right. Could it be the big cluster in T1 with low UMI and gene counts affects the sample alignment and that's the reason we are not seeing cells of T1 in the mixture cluster? Or because T1 consists of four samples? There are also some cells from the mixture cluster scattered accross the whole dataset, so maybe the mixture cells from T1 are there but also scattered accross the data. 

```{r}
DimPlot(aligned_T1_old, reduction.use='tsne', cells.highlight=aligned_T1_old@cell.names[which(aligned_T1_old@meta.data$res.0.5 %in% 'T1')], pt.size.use=0.1, cols.use = 'grey', cols.highlight = 'blue')
```

We can also look at some of the marker genes for the mixture cluster.

```{r}
T1 <- readRDS('/projects/pytrik/sc_adipose/analyze_10x_fluidigm/data/10x-180831-T1')
```

Positive markers. (the clustering was performed on the whole dataset).

Clusters 1, 2, 3, 4, 7, 11 belong to T1-T3.
Clusters 5, 6, 8, 9, 10 belong to T4-T5. 

```{r fig2, fig.height = 15, fig.width = 10, fig.align = "center"}
VlnPlot(seurobj, group.by='res.0.5', nCol=2, features.plot=c('ACTB', 'TPM2', 'ANXA2', 'LDHA', 'PRDX1', 'MYL12B', 'MYL12A', 'CCND1', 'RAB13', 'MAP1B'), point.size.use=-1)
```

Negative markers. MALAT1 and NEAT1 seem to be the best markers. 

```{r fig3, fig.height = 12, fig.width = 10, fig.align = "center"}
VlnPlot(seurobj, group.by='res.0.5', nCol=2, features.plot=c('MALAT1', 'COL1A1', 'NEAT1', 'FN1', 'COL1A2', 'HIST1H4C', 'CYR61', 'COL3A1'), point.size.use=-1)
```

It looks like cluster 11 marks the mixture cluster. The clustering here was performed on the whole dataset. Let's see where cluster 11 is:

```{r}
TSNEPlot(seurobj, group.by='res.0.5', pt.size=0.1)
```

Does cluster 11 also contain cells from other timepoints? 

```{r}
table(seurobj@meta.data$timepoint[seurobj@meta.data$res.0.5 == 11])
```

#Markergene expression

```{r}
FeaturePlot(seurobj, reduction.use='tsne18', features.plot = 'EBF2', cols.use=c('grey', 'blue'), no.legend = F)
```

```{r}
FeaturePlot(seurobj, reduction.use='tsne18', features.plot = 'TM4SF1', cols.use=c('grey', 'blue'), no.legend = F)
```

```{r}
FeaturePlot(seurobj, reduction.use='tsne18', features.plot = 'LY6K', cols.use=c('grey', 'blue'), no.legend = F)
```

```{r}
FeaturePlot(seurobj, reduction.use='tsne18', features.plot = 'PDGFRA', cols.use=c('grey', 'blue'), no.legend = F)
```
